picard/metadata.py

```diff
@@ -289,6 +294,26 @@ def compare_to_release_parts(self, release, weights):
                 except (ValueError, KeyError):
                     pass
 
+            if 'catalognumber' in self and 'catalognumber' in weights:
+                try:
+                    found = 0.0
+                    for _ in release['label-info']:
+                        if self['catalognumber'] == _['catalog-number']:
+                            print(_['catalog-number'])
+                            found += 1.0
+                            break
+                    parts.append((found, weights['catalognumber']))
+                except (KeyError, IndexError):
+                    pass
+
+            if 'barcode' in self and 'barcode' in weights:
+                score = 1.0 if self['barcode'] == release['barcode'] else 0.0
+                parts.append((score, weights['barcode']))
```

Comment:
Proper comparison of barcodes likely should be a bit more complex then exact string match. Because of UPC vs EAN-13 it is e.g. not uncommon that UPCs are written with a leading zero. Compare also e.g. https://github.com/kellnerd/harmony/blob/1744a1a87ee8a8f8a0a59cbf8208e8f01ee9bae0/utils/gtin.ts#L66